scripts/run_integration_tests.sh

```diff
@@ -152,10 +152,16 @@ sudo setcap cap_net_admin+ep target/release/vhost_user_net
 # We always copy a fresh version of our binary for our L2 guest.
 cp target/release/cloud-hypervisor $VFIO_DIR
 
+# Enable KSM with some reasonable parameters so that it won't take too long
+# for the memory to be merged between two processes.
+sudo bash -c "echo 10000 > /sys/kernel/mm/ksm/pages_to_scan"
+sudo bash -c "echo 10 > /sys/kernel/mm/ksm/sleep_millisecs"
+sudo bash -c "echo 1 > /sys/kernel/mm/ksm/run"
+
 sudo adduser $USER kvm
 newgrp kvm << EOF
 export RUST_BACKTRACE=1
-cargo test --features "integration_tests" -- --nocapture
+sudo -E PATH=$PATH bash -c 'cargo test --features "integration_tests" -- --nocapture'
```

Comment:
@sameo @rbradford 
I need your help here as I definitely don't want to run integration tests as root. Unfortunately, this is the only way that allows me to run `cat /proc/<PID>/smaps` without complaining about permissions.
I don't understand why the current user `id=sebastien`, `group=kvm` cannot read from `/proc/<PID>/smaps` since the permissions are as follows:
```
$ ll /proc/10379/smaps
-r--r--r-- 1 sebastien sebastien 0 Nov 20 13:49 /proc/10379/smaps
```
A simple `cat` fails:
```
$ cat /proc/10379/smaps
cat: /proc/10379/smaps: Permission denied
```
But adding `sudo` makes it work :(
Are there some specific capabilities I am supposed to have before I can read from this file?

Reply:
@sboeuf try CAP_SYS_PTRACE

Reply:
Or maybe....random thought, the test binary has different caps to the cloud-hypervisor binary so therefore the kernel is denying it.

Reply:
I have a solution (tested with a quick hack of just dumping but not parsing the file)

0. Firstly you must stop spawning out to bash/cat/awk as the spawned processes will not have the same caps as the test process so you will need to parse the file in Rust.
1. Build the test binary first with `--no-run` 
2. Set the cap_net_admin capability on the test binary (`sudo setcap cap_net_admin,cap_sys_ptrace+ep target/debug/deps/cloud_hypervisor-*` it will error but the right binary will be touched ) then the test binary will have the same caps and user as the cloud-hypervisor binary and so will be able to read the smaps file.

Reply:
>Firstly you must stop spawning out to bash/cat/awk as the spawned processes will not have the same caps as the test process so you will need to parse the file in Rust.

Oh so you want me to open(`/proc/<PID>/smaps`)?

>Build the test binary first with --no-run

Yes I can do that, but then how do we run the test? I would assume that `cargo test` will rebuild it, won't it?

Reply:
It won't rebuild it because the contents that would go into it have not changed.

Reply:
Ok that's great then!